woocommerce/woocommerce
# Update variation pagination table

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR includes a couple changes:
- A refactor of the Pagination component, moving it to a functional TS component and splitting the different pagination pieces into re-usuable components
- Added a `usePagination` hook to encapsulate some of the specific pagination logic
- Make use of the custom components and hook in the Variations table to align with the new design.

Closes #39885  .

https://github.com/woocommerce/woocommerce/assets/2240960/eb92aeb7-3d7e-4def-9d2f-70351ee1ec48

### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Partially Requires feature flag: `product-variation-management`**

1. Go to **Analytics > Revenue** and select the date range **Year to date**
2. Scroll down to the bottom and make sure the pagination still works correctly ( switching pages, changing the page size )
3. Enable the new product editor under **WooCommerce > Settings > Advanced > Features**
4. Install the WooCommerce Beta tester plugin and enable the `product-variation-management` feature under **Tools > WCA Test Helper > Features**
5. Go to Products > Add New
6. Add a name and save the product as draft
7. Go to **Variations** tab and add at-least one new variation option
8. Variations should be auto generated, but the pagination shouldn't show up below the Variations table ( as there should be less then 6 variations )
9. Click **Add new** for the variation options and add a couple more so it will generate at-least more then 5 ( selecting 3-4 terms per variation option )
10. It should re-generate variations
11. The pagination should now be visible, similar to this: ( picture from design )
<img width="628" alt="Screenshot 2023-08-30 at 2 33 38 PM" src="https://github.com/woocommerce/woocommerce/assets/2240960/0b198648-41b8-4871-b1a7-cf97cae4ba2e">
12. Try switching pages by the arrow or the input value, the **Viewing 1-5 of <total> items** should update accordingly.
13. Update the page size, which should display either 5, 10, or 25.
14. Note that the arrows will disappear if the total count is below the selected page size.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
